src/Access/ContextAccess.h

```diff
@@ -217,20 +223,23 @@ class ContextAccess : public std::enable_shared_from_this<ContextAccess>
 
     const AccessControl * access_control = nullptr;
     const Params params;
-    bool is_full_access = false;
-    mutable Poco::Logger * trace_log = nullptr;
-    mutable UserPtr user;
-    mutable String user_name;
-    mutable bool user_was_dropped = false;
-    mutable scope_guard subscription_for_user_change;
-    mutable std::shared_ptr<const EnabledRoles> enabled_roles;
-    mutable scope_guard subscription_for_roles_changes;
-    mutable std::shared_ptr<const EnabledRolesInfo> roles_info;
-    mutable std::shared_ptr<const AccessRights> access;
-    mutable std::shared_ptr<const AccessRights> access_with_implicit;
-    mutable std::shared_ptr<const EnabledRowPolicies> enabled_row_policies;
-    mutable std::shared_ptr<const EnabledQuota> enabled_quota;
-    mutable std::shared_ptr<const EnabledSettings> enabled_settings;
+    const bool is_full_access = false;
+
+    mutable std::atomic<bool> user_was_dropped = false;
```

Comment:
The race was because this variable was used [there](https://github.com/ClickHouse/ClickHouse/blob/3e8358e8a1be3e8efe525584df99cd50f7f8fc1c/src/Access/ContextAccess.cpp#L473) without locking the mutex.